### ISSUE TITLE
Not Working with Next.js 13 / React 18.2

### ISSUE THREAD

**Issue by travis7777** [duskload/react-device-detect#196]
1. Was working prior to Next.js 13, now getting the following error on mobile: Unhandled Runtime Error 
Error: Text content does not match server-rendered HTML.
See more info here: https://nextjs.org/docs/messages/react-hydration-error
2. Install nextjs 13, reload page that has mobile/desktop logic
3. Device user-agent - All Mobile seems
4. Device/Browser type/name - Chrome
5. Code:
import {BrowserView, MobileView, isMobile} from 'react-device-detect'

const Footer = () => {
    
    return (            
        <>
        <MobileView>              
          <>  
            Test        
          </>
        </MobileView>         
        <BrowserView>
          <>
            Test3  
          </>
        </BrowserView>  
        </>
    );
}

export default Footer;

**Comment by chris-at-lemon** [comment]
I second this. As a workaround I wrap the sniffer in useEffect to trigger it after component has loaded:

```
  const [device, setDevice] = useState<string>();
  useEffect(() => {
    if (isIOS) {
      setDevice("ios");
    }
    if (isAndroid) {
      setDevice("android");
    }
  });
```
Then use as such:

{device === "android" && <>somthing</>}

 In case it helps anyone out of a squeeze until there's a fix.

Cheers

Chris

**Comment by drbarazi** [comment]
hope this will help, i put on **app/layout.tsx**

"use client"
import '../styles/globals.css'
import { isMobile } from 'react-device-detect';
import { useEffect, useState } from 'react';

export default function RootLayout({
    children,
}: {
    children: React.ReactNode
}) {
    const [mobile, setMobile] = useState(false)
    const [isSSR, setIsSSR] = useState(true);

    useEffect(() => {
        setIsSSR(false);
        setMobile(isMobile)
    }, []);

    if (isSSR)
        return (
            <html>
                <head />
                <body>
                   Loading ....
                </body>
            </html>
        );

    return (
        <html>
            <head />
            <body>
                {mobile ?
                    <div>
                        <h1>This is rendered only on mobile</h1>
                        {children}
                    </div>
                    :
                    <div>
                            <h1>This is rendered only in browser</h1>
                            {children}
                    </div>
                }
            </body>
        </html>
    )
}

**Comment by jinojacob15** [comment]
@drbarazi  u cannot make root layout a client component .

 The statement form  next js beta docs

![DeepinScreenshot_select-area_20230306125945](https://user-images.githubusercontent.com/69616951/223045849-914e314e-5ef0-4f46-93ab-fc0abd89a6a1.png)

**Comment by theDanielJLewis** [comment]
Aside from `useEffect()`, any better workarounds or updates about Next.js app router support?

**Comment by jamesryan-dev** [comment]
+1 following

**Comment by anoop0567** [comment]
@theDanielJLewis @jamesryan-dev found any workaround for this?

**Comment by theDanielJLewis** [comment]
@anoop0567, no. I'm forced into `useEffect()` with state within a client component.

Like this:

```js
  const [device, setDevice] = useState<string>();

  useEffect(() => {
    if (isIOS) {
      setDevice("ios");
    }
    if (isMacOs) {
      setDevice("macos");
    }
    if (isAndroid) {
      setDevice("android");
    }
    if (isWindows) {
      setDevice("windows");
    }
  }, []);

return <>{device === "windows" && (<WindowsStuff />)}</>;
```

**Comment by gersooonn** [comment]
> @anoop0567, não. Sou forçado a entrar `useEffect()`no estado de um componente cliente.
> 
> Assim:
> 
> ```js
>   const [device, setDevice] = useState<string>();
> 
>   useEffect(() => {
>     if (isIOS) {
>       setDevice("ios");
>     }
>     if (isMacOs) {
>       setDevice("macos");
>     }
>     if (isAndroid) {
>       setDevice("android");
>     }
>     if (isWindows) {
>       setDevice("windows");
>     }
>   }, []);
> 
> return <>{device === "windows" && (<WindowsStuff />)}</>;
> ```
 I can't get mine to work even if I copy this code I send

**Comment by jamesryan-dev** [comment]
@anoop0567 i had to make my components client components and useState to handle any react-device-detect work

I never found a way to differentiate between tablets ( iPad ) and ios mobile devices which was a shame
